app/models/user.rb

```diff
@@ -87,9 +63,83 @@ def fetch_enrolled_clubs
     self.save!
   end
 
+  def fk_fetch_club
+    fk_authorized_clubs = Set['chemica', 'dentalia', 'filologica', 'fk', 'gbk', 'geografica', 'geologica', 'gfk', 'hermes', 'hilok', 'khk',
+                              'kmf', 'lila', 'lombrosiana', 'moeder-lies', 'oak', 'politeia', 'slavia', 'vbk', 'vdk', 'vek', 'veto',
+                              'vgk-fgen', 'vgk-flwi', 'vlak', 'vlk', 'vppk', 'vrg', 'vtk', 'wina']
+    resp = HTTParty.get("#{Rails.application.secrets.fk_auth_url}/#{username}/Gandalf",
+                        headers: {
+                          'X-Authorization' => Rails.application.secrets.fk_auth_key,
+                          'Accept' => 'application/json'
+                        })
+
+    return Set.new unless resp.success?
+
+    hash = JSON[resp.body]
+    clubs = hash['clubs'].map { |club| club['internal_name'] }.to_set
```

Comment:
If you're using this to get unique values, `.uniq` is more appropriate

Reply:
I'd like a Set as return value, not an Array